crates/sui-framework/sources/governance/sui_system.move

```diff
@@ -278,7 +278,22 @@ module sui::sui_system {
         validator::request_set_gas_price(validator, verified_cap, new_gas_price);
     }
 
-    /// A validator can call this entry function to set a new commission rate, updated at the end of the epoch.
+    /// This entry function is used to set new gas price for preactive validators
+    public entry fun set_preactive_validator_gas_price(
+        wrapper: &mut SuiSystemState,
+        new_gas_price: u64,
+        ctx: &mut TxContext,
```

Comment:
Technically preactive validator will have the operation cap too. Do we want the cap to be used here too? @longbowlu

Reply:
I have to say I don't quite understand the (new?) cap here... When using the cap, it looks like we are getting a validator address to get it from the validator set via a cap rather than using the validator's own address directly. I am not sure how good of a job I would do adding this without a deeper understanding of what's going on.

Please let me know how to proceed. If the cap is needed here, then I'd need more context. If the current solution for preactive validator gas price setting is wrong, I can also remove it for now. With this code changing quite a bit, it would be good to know what to do soon...

Reply:
Let's just do things the easiest way then, which is your current approach @awelc

Reply:
I'd suggest to stick with the Cap to avoid confusion. @awelc  your understanding is correct. Please see another comment below for using a Cap to access `&mut Validator`